plugins/mysql/mysqlembed.cpp

```diff
@@ -93,27 +93,117 @@ static void fail(const char *message)
 
 // Wrappers to MySQL structures that require corresponding releases
 
+class MySQLConnection;
+
+static __thread MySQLConnection *threadCachedConnection = nullptr;
+
+#define MAX_GLOBAL_CACHE 10
+
 class MySQLConnection : public CInterface
 {
 public:
     IMPLEMENT_IINTERFACE;
-    MySQLConnection(MYSQL *_conn) : conn(_conn)
+    MySQLConnection(MYSQL *_conn, const char *_cacheOptions, bool _threadCached, bool _globalCached) : conn(_conn), threadCached(_threadCached), globalCached(_globalCached)
     {
+        if (threadCached || globalCached)
```

Comment:
minor: I don't think _cacheOptions can be null, but it may be safer to add && _cacheOptions